.github/workflows/ci.yaml

```diff
@@ -0,0 +1,16 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up registry credentials
+      if: ${{ github.event_name != 'pull_request' }}
+      run: |
+        echo "REGISTRY_USERNAME=${{ secrets.REGISTRY_USERNAME }}" >> $GITHUB_ENV
+        echo "REGISTRY_PASSWORD=${{ secrets.REGISTRY_PASSWORD }}" >> $GITHUB_ENV
+    - name: Run container build script
+      run: bin/build_container_image
```

Comment:
Not for this PR, but I think it would be helpful to split the build into separate build and deploy steps.  This way, from GHA, we can gate the deploy separate from the build, and it's less "magical" looking as to what's happening.

Reply:
👍🏻 That sounds like a good followup since it will apply to several repos